src/ert/_c_wrappers/enkf/subst_config.py

```diff
@@ -76,33 +79,18 @@ def __init__(self, config_content=None, config_dict=None):
                 runpath_file_path,
                 "The name of a file with a list of run directories.",
             )
-
-            if ConfigKeys.NUM_CPU in config_dict:
+            if num_cpu is not None:
                 subst_list.addItem(
                     "<NUM_CPU>",
-                    str(config_dict[ConfigKeys.NUM_CPU]),
+                    str(num_cpu),
                     "The number of CPU used for one forward model.",
                 )
-            # Read num_cpu from Eclipse DATA_FILE
-            elif ConfigKeys.DATA_FILE in config_dict:
-                file_path = os.path.join(
-                    config_directory, config_dict[ConfigKeys.DATA_FILE]
-                )
-
-                if os.path.isfile(file_path) and os.access(file_path, os.R_OK):
-                    num_cpu = self._get_num_cpu(file_path)
-                    subst_list.addItem(
-                        "<NUM_CPU>",
-                        str(num_cpu),
-                        "The number of CPU used for one forward model.",
-                    )
-                else:
-                    raise IOError(f"Could not find ECLIPSE data file: {file_path}")
 
             c_ptr = self._alloc_full(subst_list)
 
         else:
-            c_ptr = self._alloc(config_content)
+            num_cpu_as_int = num_cpu if num_cpu is not None else 0
```

Comment:
Was not the default value changed to be 1 CPU?

Reply:
I see the zero as a kind of remnant of the C code / necessity from there. 0 is used here to signal "we don't have no value", which is meh, but I didn't want to work to much on the C side of subst config now, as I hope that it can be taken care of as part of the [simplify subst config issue](#4004)

the actual default used by subst config if nothing is given "from above" remains 1, as a careful study of the c code in subst_config.cpp shows ; )